packages/csharp/ArmoniK.Api.Client/EventsClientExt.cs

```diff
@@ -80,61 +83,71 @@ private static FiltersAnd ResultsFilter(string resultId)
-        if (resp.UpdateCase == EventSubscriptionResponse.UpdateOneofCase.ResultStatusUpdate && resultsNotFound.Contains(resp.ResultStatusUpdate.ResultId))
+                                                   },
+                                                   cancellationToken: cancellationToken);
+        try
         {
-          if (resp.ResultStatusUpdate.Status == ResultStatus.Completed)
+          while (await streamingCall.ResponseStream.MoveNext(cancellationToken))
           {
-            resultsNotFound.Remove(resp.ResultStatusUpdate.ResultId);
-            if (!resultsNotFound.Any())
+            var resp = streamingCall.ResponseStream.Current;
+            if (resp.UpdateCase == EventSubscriptionResponse.UpdateOneofCase.ResultStatusUpdate && resultsNotFound.Contains(resp.ResultStatusUpdate.ResultId))
             {
-              break;
-            }
-          }
+              if (resp.ResultStatusUpdate.Status == ResultStatus.Completed)
+              {
+                resultsNotFound.Remove(resp.ResultStatusUpdate.ResultId);
+                if (!resultsNotFound.Any())
+                {
+                  break;
+                }
+              }
 
-          if (resp.ResultStatusUpdate.Status == ResultStatus.Aborted)
-          {
-            throw new Exception($"Result {resp.ResultStatusUpdate.ResultId} has been aborted");
-          }
-        }
+              if (resp.ResultStatusUpdate.Status == ResultStatus.Aborted)
+              {
+                throw new ResultAbortedException($"Result {resp.ResultStatusUpdate.ResultId} has been aborted");
+              }
+            }
 
-        if (resp.UpdateCase == EventSubscriptionResponse.UpdateOneofCase.NewResult && resultsNotFound.Contains(resp.NewResult.ResultId))
-        {
-          if (resp.NewResult.Status == ResultStatus.Completed)
-          {
-            resultsNotFound.Remove(resp.NewResult.ResultId);
-            if (!resultsNotFound.Any())
+            if (resp.UpdateCase == EventSubscriptionResponse.UpdateOneofCase.NewResult && resultsNotFound.Contains(resp.NewResult.ResultId))
             {
-              break;
-            }
-          }
+              if (resp.NewResult.Status == ResultStatus.Completed)
+              {
+                resultsNotFound.Remove(resp.NewResult.ResultId);
+                if (!resultsNotFound.Any())
+                {
+                  break;
+                }
+              }
 
-          if (resp.NewResult.Status == ResultStatus.Aborted)
-          {
-            throw new Exception($"Result {resp.NewResult.ResultId} has been aborted");
+              if (resp.NewResult.Status == ResultStatus.Aborted)
+              {
+                throw new ResultAbortedException($"Result {resp.NewResult.ResultId} has been aborted");
+              }
+            }
           }
         }
+        catch (OperationCanceledException)
+        {
+        }
+        catch (RpcException)
```

Comment:
Maybe a bit too wide?

Reply:
we can improve it as we find things we nned to exclude